tap_jira/tap.py

```diff
@@ -15,33 +16,70 @@ class TapJira(Tap):
 
     config_jsonschema = th.PropertiesList(
```

Comment:
Do we want to represent the desired auth flow using `singer_sdk.typing.DiscriminatedUnion` as it was previously?

https://github.com/MeltanoLabs/tap-jira/pull/86/files#diff-9a8a5da31511510c569573541d7460963e518208e1f71162cf6847378a12dfb4L33-L48